sphinxcontrib/chapeldomain/__init__.py

```diff
@@ -596,7 +584,20 @@ def before_content(self):
         ChapelObject.before_content(self)
         if self.names:
             self.env.temp_data['chpl:class'] = self.names[0][0]
-            self.clsname_set = True
+            if not hasattr(self, 'clsname_set'):
+                self.clsname_set = 0
+            self.clsname_set += 1
+
+    def after_content(self):
+        """Called after parsing content. Pop the class name from the class
+        name
```

Comment:
I'd say "from the longer class name".  The sentence is a little tautological otherwise